### PR TITLE
refactor: feed card to experiment bookmark button

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Experiment with the card whether to include the Bookmark button or not.
- Refactor `ActionButtons` to not use `ConditionalWrapper` when unnecessary.
- Also removed the `children` props as it is not used.
- Experiment key is `bookmark_on_card`. Values is either true or false.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-2052 #done

### CODE REVIEW

File: packages/shared/src/components/Feed.tsx [hunk 1]
```diff
@@ -316,6 +316,18 @@ export default function Feed<T>({
...
+    onBookmark({
+      post,
+      origin,
+      opts: {
+        row,
+        column,
+        columns: virtualizedNumCards,
+        ...feedAnalyticsExtra(feedName, ranking),
+      },
+    });
```
Comment (rebelchris) [inline packages/shared/src/components/Feed.tsx:329]: We already have `toggleBookmark` right?
This is preferred to use in this case.

File: packages/shared/src/components/Feed.tsx [hunk 2]
```diff
@@ -376,6 +388,7 @@ export default function Feed<T>({
             user={user}
             feedName={feedName}
             ranking={ranking}
+            onBookmark={onCardBookmark}
```
Comment (rebelchris) [inline packages/shared/src/components/Feed.tsx:391]: Probably should be `toggleBookmark` to be inline

File: packages/shared/src/components/cards/ActionButtons.tsx [hunk 3]
```diff
@@ -25,10 +28,10 @@ export interface ActionButtonsProps {
   onUpvoteClick?: (post: Post) => unknown;
   onCommentClick?: (post: Post) => unknown;
   onShare?: (post: Post) => unknown;
+  onBookmark?: (post: Post) => unknown;
```
Comment (rebelchris) [inline packages/shared/src/components/cards/ActionButtons.tsx:31]: Perhaps we should keep the naming similar to `onBookmarkClick`

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice work, mainly some cleanup on naming.
And think you can re-use the existing `toggleBookmark` that's already on the feed.

I'd also like to run a e2e test locally.
wiIll do so after the quiz

File: packages/shared/src/components/Feed.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/Feed.tsx:329]: Yes, that is used here, was just renamed as `onBookmark`.

File: packages/shared/src/components/Feed.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/Feed.tsx:391]: As mentioned above, this one uses the `toggleBookmark`.

File: packages/shared/src/components/Feed.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/Feed.tsx:329]: Why did you rename it, the actual action is toggling the bookmark that seemed more explicit no?

File: packages/shared/src/components/cards/ActionButtons.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/cards/ActionButtons.tsx:31]: Cool, I based this off the `onShare` which didn't have the `Click` part, but we can do that here.

File: packages/shared/src/components/Feed.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/Feed.tsx:329]: I did not renamed it. Has been defined as an alias of that here..

File: packages/shared/src/components/cards/ActionButtons.tsx [hunk 4]
```diff
@@ -106,9 +106,8 @@ export default function ActionButtons({
   return (
     <div
       className={classNames(
-        styles.actionButtons,
-        'flex flex-row items-center',
-        insaneMode && 'justify-between',
+        'flex flex-row items-center justify-between',
+        !insaneMode && !bookmarkOnCard && 'mx-4',
```
Comment (rebelchris) [inline packages/shared/src/components/cards/ActionButtons.tsx:110]: It looks like we always need the mx-4 actually.

**Review by rebelchris — CHANGES_REQUESTED** [review]

**Review by rebelchris — APPROVED** [review]
Looks good!

**Pull request merged**